### PR TITLE
AT_077_Java | Entity > Child record loop: Delete created record

### PR DESCRIPTION
**Pull request by daria-altukhova**

### CODE REVIEW

**Review by Burdishka — APPROVED** [review]

**Comment by Burdishka** [comment]
any Trello cards?

**Review by guram21 — APPROVED** [review]

File: src/test/java/EntityChildRecordsLoopTest.java [hunk 1]
```diff
@@ -140,4 +140,50 @@ public void testEditChildRecordsLoopCard() {
...
+        WebElement targetRowDiv = findElement(By.xpath("//tbody/tr[" + numberOfCards + "]/td[4]/div[1]"));
+
+        WebElement lastCardDropdownMenu = targetRowDiv.findElement(By.xpath("button[1]"));
+        lastCardDropdownMenu.click();
+
+        WebElement deleteEntity = targetRowDiv.findElement(By.xpath("ul/li/a[text() = 'delete']"));
+        getWait().until(ExpectedConditions.visibilityOf(deleteEntity));
+        deleteEntity.click();
+
+        WebElement recycleBinIcon = findElement(By.xpath("//body/div[1]/div[2]/nav[1]/div[1]/div[2]/ul[1]/li[1]/a[1]"));
```
Comment (GalinaRuban) [inline src/test/java/EntityChildRecordsLoopTest.java:173]: such locators are not stable

File: src/test/java/EntityChildRecordsLoopTest.java [hunk 2]
```diff
@@ -140,4 +140,50 @@ public void testEditChildRecordsLoopCard() {
...
+
+        int numberOfCards = columnList.size() / 2;
+
+        double startBalanceToBeDeleted = Double.parseDouble(columnList.get(columnList.size() - 2).findElement(By.tagName("a")).getText());
+        double endBalanceToBeDeleted = Double.parseDouble(columnList.get(columnList.size() - 1).findElement(By.tagName("a")).getText());
+
+        Assert.assertEquals(startBalanceValue, startBalanceToBeDeleted);
+        Assert.assertEquals(expectedEndBalance, endBalanceToBeDeleted);
+
+        WebElement targetRowDiv = findElement(By.xpath("//tbody/tr[" + numberOfCards + "]/td[4]/div[1]"));
```
Comment (GalinaRuban) [inline src/test/java/EntityChildRecordsLoopTest.java:164]: it's ok here, but using + to concat is not the best practice in selenium. As an alternative,  the concat( ) method is used to concatenate two strings. It works similar to '+' operator.

**Review by GalinaRuban — APPROVED** [review]

File: src/test/java/EntityChildRecordsLoopTest.java (hunk 2 above)
Comment (daria-altukhova) [reply src/test/java/EntityChildRecordsLoopTest.java:164]: Thanks for the suggestion. Can you please provide any links with these selenium best practices described? My understanding is that using either `concat` method or `+` for string concatenation is equivalent from the Java perspective.

Comment (GalinaRuban) [reply src/test/java/EntityChildRecordsLoopTest.java:164]: It's based on comments from Sergei from the previous course)))

**Pull request merged**